From 26bde9cff858faef4657d81be277381dc3c2816e Mon Sep 17 00:00:00 2001
From: Helmut Grohne <helmut@subdivi.de>
Date: Sat, 17 Jun 2023 02:02:59 +0100
Subject: [PATCH] dash.postinst: fix installing with
 --path-exclude=/usr/share/man/*

The symlink might not exist, but we should remove the diversion
anyway.

Closes: #1038067
---
 debian/dash.postinst | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/debian/dash.postinst b/debian/dash.postinst
index 42b2fc9..7a1807f 100644
--- a/debian/dash.postinst
+++ b/debian/dash.postinst
@@ -64,7 +64,7 @@ drop_obsolete_diversion() {
 	dfile=$1 ltarget=$2 distrib=${3:-$dfile.distrib}
 	diverter=$(dpkg-divert --listpackage "$dfile")
 	truename=$(dpkg-divert --truename "$dfile")
-	actualtarget=$(readlink "$dfile")
+	actualtarget=$(readlink "$dfile") || actualtarget=$ltarget
 
 	if [ "$diverter" != dash ] || [ "$truename" != "$distrib" ] || [ "$actualtarget" != "$ltarget" ]; then
 		# Not our diversion or a non-trivial one.
-- 
2.39.2

